src/app/InteractionModelDelegate.h

```diff
@@ -75,6 +76,38 @@ class InteractionModelDelegate
      */
     virtual CHIP_ERROR ReportError(const ReadClient * apReadClient, CHIP_ERROR aError) { return CHIP_ERROR_NOT_IMPLEMENTED; }
 
+    /**
+     * Notification that a Command Send receive Status code embeded in Invoke Command Response
+     * @param[in]  apCommandSender A current command sender which can identify the command sender to the consumer, particularly
+     * during multiple command interactions
+     * @param[in]  aGeneralCode   Status code defined by the standard
+     * @param[in]  aProtocolId    Protocol Id
+     * @param[in]  aProtocolCode  Detailed error information, protocol-specific.
+     * @param[in]  aEndpointId    Endpoint identifier
+     * @param[in]  aClusterId     Cluster identifier
+     * @param[in]  aCommandId     Command identifier
+     * @retval # CHIP_ERROR_NOT_IMPLEMENTED if not implemented
+     */
+    virtual CHIP_ERROR CommandResponseStatus(const CommandSender * apCommandSender, const uint16_t aGeneralCode,
```

Comment:
What is the representation for a failure status response to an entire message (potentially containing potentially multiple commands in it)?  Does that just end up with `CommandResponseError` and lose the error the other side sent?

Reply:
I think we can use General Code to represent the overall error message for current IM invoke command response

Reply:
We can, but my point was: what function gets called for this case?  It can't be `CommandResponseStatus`, because there is no commandId to pass in, for example......

Reply:
Since consumer sdk know command list for current request, We can notify the current index and tell consumer sdk the status with that index, then consumer sdk can know if the command has been executed successfully or not, and instruct User next action

Reply:
in latest code, there is  commandId and mCommandIndex
            mpDelegate->CommandResponseStatus(this, generalCode, protocolId, protocolCode, endpointId, clusterId, commandId,
                                              mCommandIndex);